homeassistant/components/cover/somfy_mylink.py

```diff
@@ -0,0 +1,234 @@
+"""
+Platform for the Somfy MyLink device supporting the Synergy JsonRPC API.
+
+For more details about this platform, please refer to the documentation
+https://home-assistant.io/components/cover.somfy_mylink/
+"""
+import logging
+
+from homeassistant.components.cover import (ATTR_POSITION, SUPPORT_CLOSE,
+                                            SUPPORT_OPEN, SUPPORT_SET_POSITION,
+                                            SUPPORT_STOP, CoverDevice)
+from homeassistant.components.somfy_mylink import DATA_SOMFY_MYLINK
+from homeassistant.helpers.event import async_track_time_change
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+DEPENDENCIES = ['somfy_mylink']
+DEFAULT_SUPPORTED_FEATURES = (SUPPORT_OPEN | SUPPORT_STOP | SUPPORT_CLOSE)
+ENTITY_ID_FORMAT = 'cover.{}'
+
+
+async def async_setup_platform(hass,
+                               config,
+                               async_add_entities,
+                               discovery_info=None):
+    """Discover and configure Somfy covers."""
+    if discovery_info is None:
+        return
+    somfy_mylink = hass.data[DATA_SOMFY_MYLINK]
+    config_options = discovery_info
+    cover_list = []
+    mylink_status = await somfy_mylink.status_info()
+    for cover in mylink_status['result']:
+        cover_config = {}
+        cover_config['target_id'] = cover['targetID']
+        for key, val in config_options.get('default', {}).items():
```

Comment:
Looks like we can just copy the dict returned here. We don't need to iterate it.
```py
new = dict(old)
```

Reply:
Classic case of over-thinking things, this is much better as a copy or merge. Thanks!